account/src/main/java/com/user/account/controller/AllControlers.java

```diff
@@ -21,73 +21,79 @@
 @Validated
 @RestController
 public class AllControlers {
-	
-	
-		@Autowired
-		private AccountService accountService;
-		
-		@Autowired
-		private TransactionService transactionService;
-		
-		// tomcat test
-		@GetMapping("/home")
-		public String home() {
-			return "This is the home page";
-		}
-		
-		// Updating accountBalance after debit
-		
-		
-		@PostMapping("/debit/{accountNumber}")
-		public ResponseEntity<String> debit(@PathVariable Long accountNumber, @RequestBody Long amount) {
-			
+
+
+	@Autowired
+	private AccountService accountService;
+
+	@Autowired
+	private TransactionService transactionService;
+
+	// tomcat test
+	@GetMapping("/home")
+	public String home() {
+		return "This is the home page";
+	}
+
+	// Updating accountBalance after debit
+
+
+	@PostMapping("/debit/{accountNumber}")
+	public ResponseEntity<String> debit(@PathVariable Long accountNumber, @RequestBody String transactionAmount) {
+		try {
+
 			User user = accountService.getUser(accountNumber);
-			
+			Long amount = Long.parseLong(transactionAmount);
```

Comment:
Instead of parsing `transactionAmount` as Long datatype, obtain it as Long directly in the `debit` API, you can also have annotations to validate the transactionAmount (eg: negative value check)